index.js

```diff
@@ -1,60 +1,98 @@
 const fs = require('fs')
 const ID3Definitions = require("./src/ID3Definitions")
-const ID3Frames = require('./src/ID3Frames')
 const ID3Util = require('./src/ID3Util')
-const zlib = require('zlib')
+const ID3Helpers = require('./src/ID3Helpers')
+const { isFunction, isString } = require('./src/util')
 
 /*
 **  Used specification: http://id3.org/id3v2.3.0
 */
 
+/**
+ * Checks and removes already written ID3-Frames from a buffer
+ * @param {Buffer} data
+ * @returns {boolean|Buffer}
+ */
+function removeTagsFromBuffer(data) {
+    const framePosition = ID3Util.getFramePosition(data)
+
+    if (framePosition === -1) {
+        return data
+    }
+
+    const encodedSize = data.slice(framePosition + 6, framePosition + 10)
+    if (!ID3Util.isValidEncodedSize(encodedSize)) {
+        return false
+    }
+
+    if (data.length >= framePosition + 10) {
+        const size = ID3Util.decodeSize(encodedSize)
+        return Buffer.concat([
+            data.slice(0, framePosition),
+            data.slice(framePosition + size + 10)
+        ])
+    }
+
+    return data
+}
+
+function writeInBuffer(tags, buffer, fn) {
+    buffer = removeTagsFromBuffer(buffer) || buffer
+    const completeBuffer = Buffer.concat([tags, buffer])
+    if(isFunction(fn)) {
+        fn(null, completeBuffer)
+        return undefined
+    }
+    return completeBuffer
+}

```

Comment:
This is done differently than `read` where the buffer case is handled inside `readAync` and `readSync`, in another PR we should probably align it with `read`.

Reply:
I've changed it here because I got confused when reviewing, should be the same as read now